# Evita di richiedere NVIDIA Apex durante la sola inferenza.
#
# make_config registra anche la configurazione dell'optimizer, ma costruisce
# l'optimizer soltanto durante il training. L'import lazy mantiene invariato
# il training con FusedAdam e permette di caricare le pipeline in inferenza
# senza compilare Apex nel container ROS.

diff --git a/model/cosmos_predict2/utils/optim_instantiate_dtensor.py b/model/cosmos_predict2/utils/optim_instantiate_dtensor.py
--- a/model/cosmos_predict2/utils/optim_instantiate_dtensor.py
+++ b/model/cosmos_predict2/utils/optim_instantiate_dtensor.py
@@ -16,7 +16,6 @@
 from omegaconf import ListConfig
 from torch import nn
 
-from cosmos_predict2.utils.fused_adam_dtensor import FusedAdam
 from imaginaire.utils import log
 
 
@@ -54,6 +53,8 @@ def get_base_optimizer(
     if optim_type == "adamw":
         opt_cls = torch.optim.AdamW
     elif optim_type == "fusedadam":
+        from cosmos_predict2.utils.fused_adam_dtensor import FusedAdam
+
         opt_cls = FusedAdam
     else:
         raise ValueError(f"Unknown optimizer type: {optim_type}")
